src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sumOfAllNumbers = 0;
+  let copyOfSum;
+
+  const adder = (number = null) => {
+    sumOfAllNumbers += number;
+
+    if (number === null) {
+      copyOfSum = sumOfAllNumbers;
+      sumOfAllNumbers = 0;
+
+      return copyOfSum;
+    }
```

Comment:
There is a code style issue here. You are checking if `number` is `null` to decide whether to return the sum or the `adder` function itself. Instead of using `null` as the default parameter value and checking for it, you can make use of the fact that `undefined` is a falsy value in JavaScript. You can remove the default parameter value and check if `number` is falsy to decide whether to return the sum or the `adder` function itself.